pkg/controllermanager/controller/factory.go

```diff
@@ -62,21 +61,19 @@ import (
 
 // GardenControllerFactory contains information relevant to controllers for the Garden API group.
 type GardenControllerFactory struct {
-	cfg                    *config.ControllerManagerConfiguration
-	clientMap              clientmap.ClientMap
-	k8sGardenCoreInformers gardencoreinformers.SharedInformerFactory
-	k8sInformers           kubeinformers.SharedInformerFactory
-	recorder               record.EventRecorder
+	cfg          *config.ControllerManagerConfiguration
+	clientMap    clientmap.ClientMap
+	k8sInformers kubeinformers.SharedInformerFactory
+	recorder     record.EventRecorder
 }
 
 // NewGardenControllerFactory creates a new factory for controllers for the Garden API group.
-func NewGardenControllerFactory(clientMap clientmap.ClientMap, gardenCoreInformerFactory gardencoreinformers.SharedInformerFactory, kubeInformerFactory kubeinformers.SharedInformerFactory, cfg *config.ControllerManagerConfiguration, recorder record.EventRecorder) *GardenControllerFactory {
+func NewGardenControllerFactory(clientMap clientmap.ClientMap, kubeInformerFactory kubeinformers.SharedInformerFactory, cfg *config.ControllerManagerConfiguration, recorder record.EventRecorder) *GardenControllerFactory {
 	return &GardenControllerFactory{
-		cfg:                    cfg,
-		clientMap:              clientMap,
-		k8sGardenCoreInformers: gardenCoreInformerFactory,
-		k8sInformers:           kubeInformerFactory,
-		recorder:               recorder,
+		cfg:          cfg,
+		clientMap:    clientMap,
+		k8sInformers: kubeInformerFactory,
```

Comment:
We still have the `kubeInformerFactory` because of the specialized secret informer, correct?
I noticed, that we now have two secret informers: the specialized one from the `kubeInformerFactory` and another one for all secrets created [here](https://github.com/gardener/gardener/blob/77717eeb78fc2b98334fce0836dcadfe994aad36/pkg/controllermanager/controller/seed/seed.go#L77) / [here](https://github.com/gardener/gardener/blob/852e4ad58abb3523542f0059c5258926d02a8857/pkg/controllermanager/controller/plant/plant.go#L72).
We should probably either remove one of the informers or replace the c-r informer by a metada-only informer.

Reply:
> We should probably either remove one of the informers or replace the c-r informer by a metada-only informer.

What do you suggest concretely? Shall we simply get rid of the `k8sInformers` and only use the cr-informers now?

Reply:
That would be one option. I would just go for it and try it.
I checked the amount of secrets in one of our larger gardener installation and saw, that we have about 50MiB secrets there. So from the memory usage perspective it shouldn't be an issue to just cache all secrets. Not so sure about the implications wrt CPU usage on server/gcm side, but I would think that it's also negligible. 
Once, we vendored c-r@v0.9 we'll be able to use filtered watches for those informers again (ref https://github.com/kubernetes-sigs/controller-runtime/pull/1435)

Reply:
OK, done with ec61603b6e1d432d0ddef503c50b81e191eb58e3